### PR TITLE
MAINT: Unify polynomial fitting functions

### PR DESCRIPTION
**Pull request by eric-wieser**
These fitting functions are all the same - the algorithm used does not care about the basis.

This was done using:
* A regex find / replace on all but poly and cheb
* A manual diff showing that cheb differed only by whitespace
* A manual diff showing that poly differed in `deg.ndim == 1` vs `deg.ndim > 0`.
  Given that this function only allows `deg.ndim <= 1`, and `ndim >= 0`, these two comparison are equivalent.

---

Here's the real jackpot in terms of de-duplicated lines in `np.polynomial`

### CODE REVIEW

File: numpy/polynomial/legendre.py [hunk 1]
```diff
@@ -1452,81 +1452,7 @@ def legfit(x, y, deg, rcond=None, full=False, w=None):
...
-    # warn on rank reduction
-    if rank != order and not full:
-        msg = "The fit may be poorly conditioned"
-        warnings.warn(msg, pu.RankWarning, stacklevel=2)
-
-    if full:
-        return c, [resids, rank, s, rcond]
-    else:
-        return c
+        return pu._fit(legvander, x, y, deg, rcond, full, w)
```
Comment (mattip) [inline numpy/polynomial/legendre.py]: Needs a dedent

Comment (eric-wieser) [reply numpy/polynomial/legendre.py]: Yep, that'll teach me to always run local tests. Fixed.

**Comment by charris** [comment]
Needs rebase.

**Comment by eric-wieser** [comment]
Rebased

**Comment by charris** [comment]
Thanks Eric.

**Pull request merged**